Surface_mesh_simplification/include/CGAL/Surface_mesh_simplification/Policies/Edge_collapse/GarlandHeckbert_probabilistic_policies.h

```diff
@@ -0,0 +1,149 @@
+#include <CGAL/Surface_mesh_simplification/Policies/Edge_collapse/internal/GarlandHeckbert_core.h>
+#include <CGAL/Surface_mesh_simplification/Policies/Edge_collapse/internal/GarlandHeckbert_policy_base.h>
+
+#include <Eigen/Dense>
+
+namespace CGAL {
+namespace Surface_mesh_simplification {
+
+// forward-declare template class
+template<typename TriangleMesh, typename GeomTraits>
+class GarlandHeckbert_policies;
+
+template<typename TriangleMesh, typename GeomTraits>
+using Cost_property = CGAL::dynamic_vertex_property_t<Eigen::Matrix<typename GeomTraits::FT, 4, 4,
+  Eigen::DontAlign>>;
+
+template<typename TriangleMesh, typename GeomTraits>
+using Vertex_cost_map = typename boost::property_map<TriangleMesh, CGAL::dynamic_vertex_property_t<
+  Eigen::Matrix<typename GeomTraits::FT, 4, 4, Eigen::DontAlign>>>::type;
+
+template<typename TriangleMesh, typename GeomTraits>
+using Cost_base = internal::GarlandHeckbert_cost_base<
+    Vertex_cost_map<TriangleMesh, GeomTraits>, 
+    GeomTraits, 
+    GarlandHeckbert_policies<TriangleMesh, GeomTraits>>;
+
+template<typename TriangleMesh, typename GeomTraits>
+using Placement_base = internal::GarlandHeckbert_placement_base<
+    Vertex_cost_map<TriangleMesh, GeomTraits>, 
+    GeomTraits, 
+    GarlandHeckbert_policies<TriangleMesh, GeomTraits>>;
+
+  // derived class implements functions used in the base class that
+  // takes the derived class as template argument - see "CRTP"
+template<typename TriangleMesh, typename GeomTraits>
+class GarlandHeckbert_probabilistic_policies : 
+  public Placement_base<TriangleMesh, GeomTraits>,
+  public Cost_base<TriangleMesh, GeomTraits>
+{
+
+  public:
+
+    typedef Cost_base<TriangleMesh, GeomTraits> Get_cost;
+    typedef Placement_base<TriangleMesh, GeomTraits> Get_placement;
+
+    // these using directives are needed to choose between the definitions of these types
+    // in Cost_base and Placement_base (even though they are the same)
+    // TODO alternatives - e.g. rename base class types so they don't clash
+    using typename Cost_base<TriangleMesh, GeomTraits>::Mat_4;
+    using typename Cost_base<TriangleMesh, GeomTraits>::Col_4;
+    using typename Cost_base<TriangleMesh, GeomTraits>::Point_3;
+    using typename Cost_base<TriangleMesh, GeomTraits>::Vector_3;
+
+    typedef typename GeomTraits::FT FT;
+
+    // TODO good default values
+    GarlandHeckbert_probabilistic_policies(
+        TriangleMesh& tmesh, 
+        FT dm = FT(100),
+        Vector_3 mv = Vector_3(0.01, 0.01, 0.01), 
+        Vector_3 mn = Vector_3(0.01, 0.01, 0.01), 
+        FT sdn = 0.01,
+        FT sdp = 0.01) 
+      : Get_cost(dm), mean_vec(mv), mean_normal(mn), std_dev_normal(sdn), std_dev_pos(sdp) 
+    {
+      Vertex_cost_map<TriangleMesh, GeomTraits> vcm_ = get(Cost_property<TriangleMesh, GeomTraits>(),
+          tmesh);
+
+      /**
+       * initialize the two base class cost matrices (protected members)
+       */
+      Get_cost::m_cost_matrices = vcm_;
```

Comment:
This `vcm_` variable is going out of scope at the end of the constructor. Even if it is copied in base Cost/Placement classes, it might still not work for some mesh types.